drivers/frequency/admfm2000/README.rst

```diff
@@ -0,0 +1,452 @@
+		.baud_rate = UART_BAUDRATE,
+		.size = NO_OS_UART_CS_8,
+		.parity = NO_OS_UART_PAR_NO,
+		.stop = NO_OS_UART_STOP_1_BIT,
+		.platform_ops = UART_OPS,
+		.extra = &xuip,
+	};
+
+	struct no_os_gpio_init_param admfm2000_gpio_c0_ip[2] = {
+		{
+			.port = 0,
+			.number = 12,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+		{
+			.port = 0,
+			.number = 14,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+	};
+
+	struct no_os_gpio_init_param admfm2000_gpio_c1_ip[2] = {  
+		{
+			.port = 2,
+			.number = 10,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+		{
+			.port = 2,
+			.number = 6,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+	};
+
+	struct no_os_gpio_init_param admfm2000_gpio_dsa0_ip[5] = {  
+		{
+			.port = 2,
+			.number = 1,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+		{
+			.port = 0,
+			.number = 11,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+		{
+			.port = 2,
+			.number = 7,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+		{
+			.port = 2,
+			.number = 8,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+		{
+			.port = 2,
+			.number = 9,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+	};
+
+	struct no_os_gpio_init_param admfm2000_gpio_dsa1_ip[5] = {  
+		{
+			.port = 2,
+			.number = 20,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+		{
+			.port = 2,
+			.number = 21,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+		{
+			.port = 3,
+			.number = 8,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+		{
+			.port = 4,
+			.number = 1,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+		{
+			.port = 0,
+			.number = 13,
+			.pull = NO_OS_PULL_NONE,
+			.platform_ops = GPIO_OPS,
+			.extra = GPIO_EXTRA,
+		},
+	};
+
+	struct admfm2000_init_param admfm2000_ip = {
+		.mixer_mode = ADMFM2000_DIRECT_IF_MODE,
+		.dsa_gain = 0,
+		.gpio_sw0_param = {
+			[0] = &admfm2000_gpio_c0_ip[0],
+			[1] = &admfm2000_gpio_c0_ip[1],
+		},
+		.gpio_sw1_param = {
+			[0] = &admfm2000_gpio_c1_ip[0],
+			[1] = &admfm2000_gpio_c1_ip[1],
+		},
+		.gpio_dsa0_param = {
+			[0] = &admfm2000_gpio_dsa0_ip[0],
+			[1] = &admfm2000_gpio_dsa0_ip[1],
+			[2] = &admfm2000_gpio_dsa0_ip[2],
+			[3] = &admfm2000_gpio_dsa0_ip[3],
+			[4] = &admfm2000_gpio_dsa0_ip[4],
+		},
+		.gpio_dsa1_param = {
+			[0] = &admfm2000_gpio_dsa1_ip[0],
+			[1] = &admfm2000_gpio_dsa1_ip[1],
+			[2] = &admfm2000_gpio_dsa1_ip[2],
+			[3] = &admfm2000_gpio_dsa1_ip[3],
+			[4] = &admfm2000_gpio_dsa1_ip[4],
+		},
```

Comment:
I think you could write these assignments as  `.gpio_dsa1_param = admfm2000_gpio_dsa1_ip`